### DISCUSSION TITLE
Reading headers from the Response for HTTP Connect method to the Proxy Server

### DISCUSSION THREAD

**Discussion by sergzak022** [TooTallNate/proxy-agents#310] [Q&A] [answered]
Hello,

When I send a request through a proxy server using `curl` with `--verbose` option it also prints headers that come from the Response for HTTP Connect method to the Proxy Server. Does `HttpsProxyAgent` provides a way to read these headers?

Just to clarify, I'm not interested in headers from the target server, but interested in headers from the Proxy Server.

Thank you!

**Comment by TooTallNate** [comment, marked as answer]
You can listen for the `"proxyConnect"` event on either the `Request` instance or the `Agent` instance.

**Comment by sergzak022** [comment]
Thank you for quick reply, @TooTallNate! Sorry, didn't realize I could use discussions. I won't be polluting your issues anymore with similar questions.
